src/integrations/watchers/indexable-permalink-watcher.php

```diff
@@ -161,7 +181,19 @@ public function force_reset_permalinks() {
 	 * @return bool Whether the permalinks should be reset.
 	 */
 	public function should_reset_permalinks() {
-		return \get_option( 'permalink_structure' ) !== $this->options_helper->get( 'permalink_structure' );
+		if ( \get_option( 'permalink_structure' ) !== $this->options_helper->get( 'permalink_structure' ) ) {
+			$this->reason = Indexing_Notification_Integration::REASON_PERMALINK_SETTINGS;
+			return true;
+		}
+		elseif ( \get_option( 'category_base' ) !== $this->options_helper->get( 'category_base_url' ) ) {
+			$this->reason = Indexing_Notification_Integration::REASON_CATEGORY_BASE_PREFIX;
+			return true;
+		}
+		elseif ( \get_option( 'tag_base' ) !== $this->options_helper->get( 'tag_base_url' ) ) {
+			$this->reason = Indexing_Notification_Integration::REASON_TAG_BASE_PREFIX;
+			return true;
+		}
+		return false;
```

Comment:
Unit tests fail, not 100% sure why, but seems to me setting the "reason" in this "should" method is part of the problem as the actual "reason" value in the tests is `null`. Also, I'm seeing that in the tests the expected subtype and the actual subtype are different.

Reply:
The problem indeed lied within the reason property, fixed in last commit